### DISCUSSION TITLE
SignIn with OTP is not perfect

### DISCUSSION THREAD

**Discussion by dimad4441** [supabase/supabase#14341] [Questions]
On my flutter dev, I remember that any signing process with OTP doesn't work as we want. I try to sign in with phone and also with magic link but after sign in success we cannot check the current logged user.

```dart
currentSession return null
currentUser return null
onAuthStateChange is not fire
```

**Comment by souvik666** [comment]
You can find an example app [here](https://supabase.com/docs/guides/with-flutter)!
